addons/l10n_ec/migrations/3.5/post-migration.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, SUPERUSER_ID
+
+def update_withhold_income_sale_repartition_lines(env):
+    # Delete tag_ids for withhold income sale repartition lines
+    # Delete tag_ids for withhold income sale account move lines
+    report_tax_codes = ['343', '393', '346', '396', '344', '394', '3440', '3940', '345', '395']
+    account_account_tags_names = []
+    for tax_code in report_tax_codes:
+        account_account_tags_names.append(f'+{tax_code} (Reporte 103)')
+        account_account_tags_names.append(f'-{tax_code} (Reporte 103)')
+    account_account_tags = env['account.account.tag'].search([('name', 'in', account_account_tags_names)])
+    tax_group_id = env.ref('l10n_ec.tax_group_withhold_income_sale')
+    domain_repartition_line = ['|', ('invoice_tax_id.tax_group_id', '=', tax_group_id.id), ('refund_tax_id.tax_group_id', '=', tax_group_id.id)]
+    repartition_line_ids = env['account.tax.repartition.line'].search(domain_repartition_line)
+    account_move_line_ids = env['account.move.line'].search([('tax_line_id.tax_group_id', '=', tax_group_id.id)])
+    query = '''
+    DELETE FROM account_account_tag_account_tax_repartition_line_rel 
+           WHERE account_tax_repartition_line_id IN {repartition_line_ids}
+           AND account_account_tag_id IN {account_account_tags_ids};
```

Comment:
with the IN, the entire idea of using SQL gets lost.

Reply:
I think that using `IN` is fine in general, we do that all the time in migration scripts ([example](https://github.com/odoo/upgrade/pull/4495/files#diff-def617aa3a377c2bcacdf82e543ab4317c24aa381a21a13c71228c1a9b129289R84-R87)). what we want to improve is the way values are inserted into the queries: do not use `format`/`f-strings`/similars as they may introduce an injection vulnerability in the code. Let, instead the [cursor](https://www.psycopg.org/docs/cursor.html#cursor.execute) do its job by deciding when to put the variables in. Small example:

```
ids = [1,2,3]

cr.execute("DELETE FROM table_name WHERE id IN %s", [tuple(ids)])
```